### PR TITLE
Upload donors

### PR DESCRIPTION
**Pull request by DGitRich**

### CODE REVIEW

File: Unite.Web/Client/public/templates/donors.tsv [hunk 1]
```diff
@@ -0,0 +1,2 @@
+id	mta	projects	studies	sex	age	diagnosis	diagnosis_date	primary_site	localization	vital_status	vital_status_change_date	vital_status_change_day	progression_status	progression_status_change_date	progression_status_change_day	kps_baseline	steroids_baseline
+DO1	mta	WP1	ST1	sex	56	Glioblastoma	2020-01-01	Brain	Left	true	2021-01-01	365	false	2020-02-12	37	90	false
```
Comment (vulasau) [inline Unite.Web/Client/public/templates/donors.tsv:2]: 'mta' value here should be true, false or null.

File: Unite.Web/Client/src/domain/_shared/components/upload/UploadDialog.vue [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+          dense flat no-caps
+          v-close-popup
+        />
+      </q-card-actions>
+    </q-card>
+  </q-dialog>
+</template>
+
+<script>
+const defaultErrorMessage = 'Something went terribly wrong, sorry!';
```
Comment (vulasau) [inline Unite.Web/Client/src/domain/_shared/components/upload/UploadDialog.vue:62]: I like the message, but we should probably write something less horrifying for users :)

File: Unite.Web/Client/src/domain/donors/api/index.js [hunk 3]
```diff
@@ -1,15 +1,15 @@
-import settings from "@/settings";
-import DomainApi from "@/domain/_shared/api/domain-api";
+import settings from '@/settings';
```
Comment (vulasau) [inline Unite.Web/Client/src/domain/donors/api/index.js:1]: Use double quotes "" wherever and whenever possible and single quotes '' for interpolation between "" e.g. "Values is 'hello'";

File: Unite.Web/Client/src/domain/donors/components/UploadButton.vue [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+  <q-btn
+    label="Upload"
+    title="Upload data"
+    icon="las la-file-upload"
+    color="primary"
+    dense flat no-caps>
+    <q-menu>
+      <q-list>
+        <q-item @click="showUploadDonors" clickable v-close-popup dense>
```
Comment (vulasau) [inline Unite.Web/Client/src/domain/donors/components/UploadButton.vue:26]: show*** is confusing name here. Usually such names like "show", "hide", "display" are associated with display events - whether or not the components should be visible in the screen. ShowUpload* also shows or hides dialog, but I would call it as typical click event with "on" prefix - onUploadDonors, onUploadTreatments. Then inside them we can update display properties for dialogs and actual methods to fetch data from database are uploadDonors, uploadTreatments. So "on*" - event responses on the UI.

File: Unite.Web/Client/src/domain/_shared/components/upload/UploadDialog.vue (hunk 2 above)
Comment (DGitRich) [reply Unite.Web/Client/src/domain/_shared/components/upload/UploadDialog.vue]: Removed this error as it was obsolete

**Review by vulasau — APPROVED** [review]

**Pull request merged**